src/rev_ai/speechrec.py

```diff
@@ -9,67 +9,85 @@
 import requests
 import logging
 import json
+from models import *
 
 LOG = logging.getLogger(__name__)
 
+class RevSpeechAPIClient:
+    """Client which implements rev.ai API"""
+
+    VERSION = "v1"
+    """Default version of rev.ai"""
 
-class RevSpeechAPI:
-    VERSION = "v1beta"
     BASE_URL = "https://api.rev.ai/revspeech/{v}/".format(v=VERSION)
+    """Default address of the API"""
+
+    def __init__(
+        self,
+        api_key
+    ):
+        """
+        Constructor
 
-    def __init__(self, api_key, use_stage=False):
+        :param api_key: api key which authorizes all requests and links them to your account.
+            Generated on the settings page of your account dashboard on rev.ai
+        """
         if not api_key:
             raise ValueError("API Key cannot be empty.")
         self.s = requests.Session()
         self.s.headers.update({
             'Authorization': 'Bearer {api_key}'.format(api_key=api_key)
         })
-        if use_stage:
-            self.BASE_URL = ("https://api-stage.rev.ai/revspeech/{v}/"
-                             .format(v=self.VERSION))
 
 
-    def submit_job_url(self, media_url, metadata="", callback_url=None):
+    def submit_job_url(
```

Comment:
question, should we have 2 urls for url/local file? 

similar for get _transcript, I would think we should have 1 method and allow the user to specify the type output which is a enum/string which is code somewhere.

I don't feel very strongly about submit_job, but I think get_transcript should be one method (which gives these methods a 1-1 relationship with available endpoints

Reply:
I would say two methods is preferable to the above. I would avoid function signatures where we want the user to set `media_url` XOR `file`. Another option is to 'smartly detect' if the provided media is a file on the system. But my most preferred is with two separate functions.

Reply:
Yeah for submit job, it makes sense to have 2.

I would have 1 for get transcript though

Reply:
I think since get transcript has multiple return types it makes sense to have 2 methods. Yes this breaks the 1:1 endpoint:method relationship we had with the API but I think the SDK if more clear because of it